geniza/corpus/models.py

```diff
@@ -434,28 +451,78 @@ def editions(self):
     @classmethod
     def items_to_index(cls):
         """Custom logic for finding items to be indexed when indexing in
-        bulk; only include public docs."""
-        return cls.objects.filter(status=Document.PUBLIC)
+        bulk."""
+        # TODO: can we share common/reused prefetching logic
+        # in a custom qureyset filter or similar? (adapted here from admin)
+        return cls.objects.select_related("doctype").prefetch_related(
+            "tags",
+            "languages",
+            "footnotes",
+            "log_entries",
+            Prefetch(
+                "textblock_set",
+                queryset=TextBlock.objects.select_related(
+                    "fragment", "fragment__collection"
+                ),
+            ),
+        )
```

Comment:
Love the idea to prefetch/optimize these index-time queries, and would be great to see if we can extract common parts as queryset filters!